### PR TITLE
Deprecate global `Homebrew.args`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Finished the implementation started by @SeekingMeaning in https://github.com/Homebrew/brew/pull/8204.

Closes https://github.com/Homebrew/brew/pull/8204.

### CODE REVIEW

**Comment by SeekingMeaning** [comment]
For some reason this PR (and my old one) doesn't work with https://github.com/Homebrew/homebrew-livecheck/commit/aafb126ada9c0625929aa4406532021a83ca8a4b

File: Library/Homebrew/compat/dependencies_helpers.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  module Compat
+    @printed_includes_ignores_warning = false
+
+    def argv_includes_ignores(_argv)
+      unless @printed_includes_ignores_warning
+        odeprecated "Homebrew.argv_includes_ignores", "Homebrew.args_includes_ignores"
+        @printed_includes_ignores_warning = true
+      end
+      args_includes_ignores(Homebrew.args)
+    end
```
Comment (SeekingMeaning) [inline Library/Homebrew/compat/dependencies_helpers.rb:13]: Should be noted that [`beeftornado/rmtree`](https://github.com/beeftornado/homebrew-rmtree) explicitly passes arguments to `argv_includes_ignores`:

https://github.com/beeftornado/homebrew-rmtree/blob/a26a861/cmd/brew-rmtree.rb#L176

(Something to consider in terms of backwards compatibility)

Comment (reitermarkus) [reply Library/Homebrew/compat/dependencies_helpers.rb:13]: The way it is used in `rmtree`, using

```
Homebrew.argv_includes_ignores(["--installed"])
```

doesn't even return anything with the old version.

Comment (reitermarkus) [reply Library/Homebrew/compat/dependencies_helpers.rb:13]: The way it is used in `rmtree`, using

```
Homebrew.argv_includes_ignores(["--installed"])
```

it always returns the same thing.

**Comment by MikeMcQuaid** [comment]
Thanks @reitermarkus ❤️

**Pull request merged**